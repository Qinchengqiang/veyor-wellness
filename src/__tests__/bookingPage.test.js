import {cleanup, render, screen} from '@testing-library/react';  // testing-library
import renderer from 'react-test-renderer';
import React from 'react';
import BookingPage from "../components/bookingSystem/bookingPage";

const AppWrapper = () => {

    return (
        <BookingPage/>
    )
}

afterEach(() => {
    cleanup();
});

describe('App group:', () => {
    // test 1
    it('App test 1: render newBooking page', () => {
        render(<AppWrapper/>);
        const BookEle = screen.getByTestId('Booking');
        expect(BookEle).toBeInTheDocument();
    })

    // test 2
    test('App test 2: title', () => {
        render(<AppWrapper/>);
        const linkElement = screen.getByText(/Book a wellness session/i);
        expect(linkElement).toBeInTheDocument();
    });

})


/**
 * snapshot test (using react-test-renderer)
 */
test('App test: matches snapshot', () => {
    const tree = renderer.create(<AppWrapper/>).toJSON();
    expect(tree).toMatchSnapshot();                 // failed，press ‘u’ to update. (jest -u)
})