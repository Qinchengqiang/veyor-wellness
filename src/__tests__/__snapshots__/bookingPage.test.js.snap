// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test: matches snapshot 1`] = `
<div
  className="newBooking h-100 d-flex flex-column align-items-center"
>
  <h1
    data-testid="Booking"
  >
    Book a wellness session.
  </h1>
  <p
    className="mt-4"
  >
    Visit one of our expert consultant to get yourself feeling 100% again.
  </p>
  <div
    className="tabs mt-4 d-flex flex-row align-items-stretch"
  >
    <div
      className="tab h-100 p-2 border-left-radius3 d-flex flex-row align-items-center zIndex2  tab-active"
    >
      <div
        className="tabLeft"
      >
        Choose Appointment
      </div>
      <div
        className="tabTriangle"
      />
    </div>
    <div
      className="tab h-100 p-2 d-flex flex-row align-items-center border-left-0 zIndex1"
    >
      <div
        className="tabLeft paddingLeft"
      >
        Your info
      </div>
      <div
        className="tabTriangle"
      />
    </div>
    <div
      className="tab h-100 p-2 d-flex flex-row align-items-center border-left-0 border-right-radius3"
    >
      <div
        className="tabLeft paddingLeft"
      >
        Confirmation
      </div>
    </div>
  </div>
  <p>
    Step 
    1
     
  </p>
  <form
    onSubmit={[Function]}
  />
</div>
`;
